src/game/common/system/geometry.cpp

```diff
@@ -270,30 +279,37 @@ void GeometryInfo::Get_2D_Bounds(const Coord3D &pos, float angle, Region2D &regi
             region.hi.y = pos.y + m_majorRadius;
             break;
         case GEOMETRY_BOX: {
-            float sin_theta = Sin(angle);
             float cos_theta = Cos(angle);
+            float sin_theta = Sin(angle);
+
             float adj_major = cos_theta * m_majorRadius;
             float adj_minor = cos_theta * m_minorRadius;
+
             float opp_major = sin_theta * m_majorRadius;
             float opp_minor = sin_theta * m_minorRadius;
-            region.lo.x = region.hi.x = float(pos.x - adj_major) - opp_minor;
-            region.lo.y = region.hi.y = float(pos.y + adj_minor) - opp_major;
 
-            float tmp_x = float(adj_major + pos.x) - opp_minor;
-            float tmp_y = float(opp_major + pos.y) + adj_minor;
+            float tmp_x = float(pos.x - adj_major) - opp_minor;
+            float tmp_y = float(pos.y + adj_minor) - opp_major;
```

Comment:
These float casts are pointless all these calcs are with float's and floats are always converted to.